### DISCUSSION TITLE
Daten vom HM-1500 werden nicht aktualisiert

### DISCUSSION THREAD

**Discussion by killerkasi** [tbnobody/OpenDTU#673] [Q&A]
Hallo zusammen,

ich habe seit ein paar Tagen das Problem, das die Daten von meinem HM-1500 nicht alle 10 Sekunden aktualisert werden.
Es varieirt stark. Manchmal nach 30 Sekunden, manchmal erst nach 1 -2 Minuten.
Es ist dann so, das der HM-1500 in der Live-Übersicht in "gelb" und wenn es zu lange ist, sogar in "rot" dargestellt wird.
Manchmal werden alle Werte nur noch mit 0 angezeigt.

Ich habe noch einen HM-400 eingebunden, der macht dieses Problem nicht.

Ich werden mit dem Blick in die Konsole auch nicht wirklich schlau. Aber es sieht so aus, als wenn es Fehler gibt, bzw er nicht senden oder Empfangen kann.

Wenn der HM-1500 Abgefragt wird, findet man folgendes in der Konsole:

11:33:04.004 > Fetch inverter: 11617440XXXX
11:33:04.066 > TX RealTimeRunData Channel: 3 --> 15 74 40 15 21 80 18 14 40 80 0B 00 64 08 64 5F 00 00 00 00 00 00 00 00 7D 05 7D 
11:33:04.125 > Interrupt received
11:33:04.233 > RX Channel: 40 --> 95 74 40 15 21 74 40 15 21 01 00 01 01 63 00 BE 00 BE 02 A3 02 A1 00 01 62 94 02 
11:33:04.279 > Interrupt received
11:33:04.332 > RX Channel: 40 --> 95 74 40 15 21 74 40 15 21 02 00 01 92 24 00 61 00 5F 01 64 00 BC 00 BC 02 9D E4 
11:33:04.434 > Interrupt received
11:33:04.482 > RX Channel: 23 --> 95 74 40 15 21 74 40 15 21 03 02 9D 00 00 93 32 00 00 B9 81 00 5F 00 60 09 17 B1 
11:33:04.532 > RX Period End
11:33:04.532 > Last missing
11:33:04.532 > Request retransmit: 4
11:33:04.532 > TX RequestFrame Channel: 23 --> 15 74 40 15 21 80 18 14 40 84 5D 
11:33:04.641 > RX Period End
11:33:04.641 > Last missing
11:33:04.641 > Request retransmit: 4
11:33:04.641 > TX RequestFrame Channel: 40 --> 15 74 40 15 21 80 18 14 40 84 5D 
11:33:04.684 > RX Period End
11:33:04.684 > Last missing
11:33:04.684 > Request retransmit: 4
11:33:04.684 > TX RequestFrame Channel: 61 --> 15 74 40 15 21 80 18 14 40 84 5D 
11:33:04.840 > Interrupt received
11:33:04.948 > RX Channel: 3 --> 95 74 40 15 21 74 40 15 21 84 13 89 09 F8 00 D3 00 6E 03 E5 00 59 0C 1B D7 25 9D 
11:33:05.053 > RX Period End
11:33:05.053 > Success
11:33:05.088 > TX AlarmData Channel: 75 --> 15 74 40 15 21 80 18 14 40 80 11 00 64 08 64 5F 00 00 00 00 00 00 00 00 A7 1E A6 
11:33:05.136 > Interrupt received
11:33:05.259 > RX Channel: 40 --> 95 74 40 15 21 74 40 15 21 01 00 01 80 01 00 01 51 17 51 17 00 00 00 00 80 02 97 
11:33:05.306 > Interrupt received
11:33:05.381 > RX Channel: 75 --> 95 74 40 15 21 74 40 15 21 03 86 1C 00 00 0E 15 80 02 0C 0F 86 25 86 25 FF FF 96 
11:33:05.461 > Interrupt received
11:33:05.512 > RX Channel: 61 --> 95 74 40 15 21 74 40 15 21 06 86 27 00 00 0E 14 80 02 0C 13 86 2F 86 2F FF FF B5 
11:33:05.663 > Interrupt received
11:33:05.715 > RX Channel: 23 --> 95 74 40 15 21 74 40 15 21 09 86 34 00 00 0E 16 80 02 0C 17 86 3A 86 3A FF FF AF 
11:33:05.865 > RX Period End
11:33:05.865 > Last missing
11:33:05.865 > Request retransmit: 10
11:33:05.865 > TX RequestFrame Channel: 3 --> 15 74 40 15 21 80 18 14 40 8A 53 
11:33:05.919 > Interrupt received
11:33:05.978 > RX Channel: 61 --> 95 74 40 15 21 74 40 15 21 0A F1 ED 80 02 0C 18 86 3A 86 3A 00 00 0E 13 80 02 8A 
11:33:06.026 > RX Period End
11:33:06.026 > Last missing
11:33:06.026 > Request retransmit: 11
11:33:06.026 > TX RequestFrame Channel: 23 --> 15 74 40 15 21 80 18 14 40 8B 52 
11:33:06.072 > Interrupt received
11:33:06.125 > RX Channel: 75 --> 95 74 40 15 21 74 40 15 21 0B 0C 19 86 3A 86 3A FF FF F1 ED 80 02 0C 1A 86 3C B9 
11:33:06.279 > RX Period End
11:33:06.279 > Last missing
11:33:06.279 > Request retransmit: 12
11:33:06.279 > TX RequestFrame Channel: 40 --> 15 74 40 15 21 80 18 14 40 8C 55 
11:33:06.381 > Interrupt received
11:33:06.434 > RX Channel: 75 --> 95 74 40 15 21 74 40 15 21 8C 86 3C 00 00 0E 13 73 14 D9 
11:33:06.585 > RX Period End
11:33:06.585 > Middle missing
11:33:06.585 > Request retransmit: 2
11:33:06.585 > TX RequestFrame Channel: 61 --> 15 74 40 15 21 80 18 14 40 82 5B 
11:33:06.593 > Interrupt received
11:33:06.654 > RX Channel: 3 --> 95 74 40 15 21 74 40 15 21 02 0C 0D 86 1B 86 1B FF FF F1 EC 80 02 0C 0E 86 1C 91 
11:33:06.785 > RX Period End
11:33:06.785 > Middle missing
11:33:06.785 > Request retransmit: 4
11:33:06.785 > TX RequestFrame Channel: 75 --> 15 74 40 15 21 80 18 14 40 84 5D 
11:33:06.835 > RX Period End
11:33:06.835 > Middle missing
11:33:06.835 > Retransmit timeout

Kann mir jemand sagen, ob das was in der Konsole ausgeworfen ist, so OK ist ?
Hat noch jemand das Problem was ich oben beschrieben habe?

Vielen Dank für Eure Hilfe

**Comment by Fireblade900rr** [comment]
Ich habe genau das gleiche Problem, ich meine vor der Firmware [055eb3a](https://github.com/tbnobody/OpenDTU/commits/055eb3a) lief es normal, nun etliche fehlgeschlagene Kommunikationsversuche.

**Comment by tbnobody** [comment]
Ich versuche mal die verschiedenen Zeilen zu erklären die man im Log oben sieht:


```
### Anforderung von Eventlog Daten:
11:33:05.088 > TX AlarmData Channel: 75 --> 15 74 40 15 21 80 18 14 40 80 11 00 64 08 64 5F 00 00 00 00 00 00 00 00 A7 1E A6 
### Paket Empfangen
11:33:05.136 > Interrupt received
### Paket aus dem NRF Modul gelesen --> Paket 1
11:33:05.259 > RX Channel: 40 --> 95 74 40 15 21 74 40 15 21 01 00 01 80 01 00 01 51 17 51 17 00 00 00 00 80 02 97
11:33:05.306 > Interrupt received
### Paket aus dem NRF Modul gelesen --> Paket 3
11:33:05.381 > RX Channel: 75 --> 95 74 40 15 21 74 40 15 21 03 86 1C 00 00 0E 15 80 02 0C 0F 86 25 86 25 FF FF 96
11:33:05.461 > Interrupt received
### Paket aus dem NRF Modul gelesen --> Paket 6
11:33:05.512 > RX Channel: 61 --> 95 74 40 15 21 74 40 15 21 06 86 27 00 00 0E 14 80 02 0C 13 86 2F 86 2F FF FF B5
11:33:05.663 > Interrupt received
### Paket aus dem NRF Modul gelesen --> Paket 9
11:33:05.715 > RX Channel: 23 --> 95 74 40 15 21 74 40 15 21 09 86 34 00 00 0E 16 80 02 0C 17 86 3A 86 3A FF FF AF
### Empfangszeit abgelaufen
11:33:05.865 > RX Period End
### Es wurde kein Paket mit den Bits 0x80 gefunden --> Das letzte Paket muss fehlen
11:33:05.865 > Last missing
### Fordere Paket 10 an
11:33:05.865 > Request retransmit: 10
11:33:05.865 > TX RequestFrame Channel: 3 --> 15 74 40 15 21 80 18 14 40 8A 53
11:33:05.919 > Interrupt received
### Paket aus dem NRF Modul gelesen --> Paket 10 (0x0A)
11:33:05.978 > RX Channel: 61 --> 95 74 40 15 21 74 40 15 21 0A F1 ED 80 02 0C 18 86 3A 86 3A 00 00 0E 13 80 02 8A
### Empfangszeit abgelaufen
11:33:06.026 > RX Period End
### Es wurde kein Paket mit den Bits 0x80 gefunden --> Das letzte Paket muss fehlen
11:33:06.026 > Last missing
### Fordere Paket 11 an
11:33:06.026 > Request retransmit: 11
11:33:06.026 > TX RequestFrame Channel: 23 --> 15 74 40 15 21 80 18 14 40 8B 52
11:33:06.072 > Interrupt received
### Paket aus dem NRF Modul gelesen --> Paket 11 (0x0B)
11:33:06.125 > RX Channel: 75 --> 95 74 40 15 21 74 40 15 21 0B 0C 19 86 3A 86 3A FF FF F1 ED 80 02 0C 1A 86 3C B9
### Empfangszeit abgelaufen
11:33:06.279 > RX Period End
### Es wurde kein Paket mit den Bits 0x80 gefunden --> Das letzte Paket muss fehlen
11:33:06.279 > Last missing
### Fordere Paket 11 an
11:33:06.279 > Request retransmit: 12
11:33:06.279 > TX RequestFrame Channel: 40 --> 15 74 40 15 21 80 18 14 40 8C 55
11:33:06.381 > Interrupt received
### Paket aus dem NRF Modul gelesen --> Paket 12 (0x8C) --> Damit wurde das letzte Paket gefunden (0x80)
11:33:06.434 > RX Channel: 75 --> 95 74 40 15 21 74 40 15 21 8C 86 3C 00 00 0E 13 73 14 D9
### Empfangszeit abgelaufen
11:33:06.585 > RX Period End
### Weiter oben wurde aber z.B. das 2er Paket nicht empfangen, daher fehlt ein Paket in der Mitte
11:33:06.585 > Middle missing
### Fordere Paket 2 an
11:33:06.585 > Request retransmit: 2
11:33:06.585 > TX RequestFrame Channel: 61 --> 15 74 40 15 21 80 18 14 40 82 5B
11:33:06.593 > Interrupt received
### Paket aus dem NRF Modul gelesen --> Paket 2 (0x02)
11:33:06.654 > RX Channel: 3 --> 95 74 40 15 21 74 40 15 21 02 0C 0D 86 1B 86 1B FF FF F1 EC 80 02 0C 0E 86 1C 91
11:33:06.785 > RX Period End
### Weiter oben wurde aber z.B. das 4er Paket nicht empfangen, daher fehlt ein Paket in der Mitte
11:33:06.785 > Middle missing
11:33:06.785 > Request retransmit: 4
11:33:06.785 > TX RequestFrame Channel: 75 --> 15 74 40 15 21 80 18 14 40 84 5D
11:33:06.835 > RX Period End
11:33:06.835 > Middle missing
11:33:06.835 > Retransmit timeout
```

Es werden also durchaus Daten empfangen, aber beim ersten lesen wurden nicht alle Pakete empfangen. Dadurch werden diese nach und nach neu angefordert. Kann natürlich auch an einem schlechten Empfang liegen. Oder mangelhafte Stromversorgung des NRF Moduls (Kondensator verbaut?) oder oder oder...

**Reply by Fireblade900rr** [reply]
Hey, Danke für die Antwort, ich habe den "Fehler" durch Zufall gerade bei mir gefunden. Ich habe die Sendeleistung wieder auf das Minimum (-18dbm) reduziert, seitdem läuft die Kommunikation wieder wie geschmiert. Wo könnte da der Fehler liegen ?

Nun habe ich immer diesen Block in der Konsole, so wie es sich gehört.

16:03:11.882 > Fetch inverter: xxxxxxxxxx
16:03:11.944 > TX RealTimeRunData Channel: 23 --> 15 83 12 54 73 80 14 63 72 80 0B 00 64 0B 46 AF 00 00 00 00 00 00 00 00 9D AE 18 
16:03:12.040 > Interrupt received
16:03:12.090 > RX Channel: 75 --> 95 83 12 54 73 83 12 54 73 01 00 01 01 49 00 A6 00 BB 02 21 02 67 00 00 02 BF 3B 
16:03:12.245 > Interrupt received
16:03:12.292 > RX Channel: 61 --> 95 83 12 54 73 83 12 54 73 02 00 00 03 15 00 76 00 86 00 0C 00 02 00 01 00 00 7E 
16:03:12.354 > Interrupt received
16:03:12.406 > RX Channel: 61 --> 95 83 12 54 73 83 12 54 73 03 00 00 00 00 00 00 00 00 00 00 00 00 00 00 09 60 FF 
16:03:12.457 > Interrupt received
16:03:12.507 > RX Channel: 40 --> 95 83 12 54 73 83 12 54 73 84 13 86 04 4E 00 E1 00 2E 03 D4 00 81 00 12 08 E3 AE 
16:03:12.566 > RX Period End
16:03:12.566 > Success

**Reply by killerkasi** [reply]
Vielen Dank für die Antworten.
Ja es scheint tatsächlich so zu sein, dass es am Empfang liegt.
Ich habe die Herausforderung, dass ich die DTU zentral im Haus positionieren muss, damit ich den WR vom Dach und den aus dem Keller (zur Nachteinspeisung) erreiche.
Mich wundert nur stark. Das der aus dem Keller, gefühlt, mehr Strecke mit Hindernissen wie Decke und Wänden überwinden kann.

Zur Antenne:
Ich habe so eine verbaut: https://amzn.eu/d/11qOCjM

Gibt es eigentlich schon eine Idee / Möglichkeit das eine weitere openDTU als Repeater eingesetzte werden kann?

**Reply by tbnobody** [reply]
> Gibt es eigentlich schon eine Idee / Möglichkeit das eine weitere openDTU als Repeater eingesetzte werden kann?

Ein Repeater Einsatz ist so nicht vorgesehen (und nichts was ich implementieren möchte). Du kannst aber einfach mehrere ESP's mit nRF24 Modulen in den gleichen MQTT Broker schreiben lassen und dort zentral auswerten.

**Reply by Ezeqeel13** [reply]
> Hey, Danke für die Antwort, ich habe den "Fehler" durch Zufall gerade bei mir gefunden. Ich habe die Sendeleistung wieder auf das Minimum (-18dbm) reduziert, seitdem läuft die Kommunikation wieder wie geschmiert. Wo könnte da der Fehler liegen ?
> 
Der Thread ist zwar schon älter, aber trotzdem danke für den Tipp. Nach einem Reboot der openDTU musste ich die Sendeleistung auch reduzieren. Stand auf 0 dbm, musste ich auf den kleinsten Wert -10dbm setzen. Seither ist es wieder stabil.

**Comment by tbnobody** [comment]
> Ich habe die Sendeleistung wieder auf das Minimum (-18dbm) reduziert, seitdem läuft die Kommunikation wieder wie geschmiert. Wo könnte da der Fehler liegen ?

Bei Funkwellen ist es wie mit Wasser (doofer vergleich, aber irgendwie schon). Wenn es große Wellen (hohe Sendeleistung) auf kleinerem Raum gibt (oder der Empfänger noch zu nahe dran ist) werden die Wellen ggf. reflektiert oder überlagern sich. Dadurch kann es sein das die Signale nicht mehr gelesen werden können. Desweiteren benötigen größere Wellen mehr Power zum erzeugen. Wenn die Spannungsversorung das nicht hergibt können sich die Wellen verformen bzw. werden "verstümmelt" gesendet und können daher nicht mehr korrekt interpretiert werden.
